[CMS-799][BUGS-5125] Release notes for pantheon-cache WPCLI command deprecation, REST API responses (#7599)
pantheon-systems/documentation: source/content/start-states/wordpress.md

## Latest Release

> ### 2022-08-16

Reviewer: I imagine we'll hold this update for the next WP core release, so this date will likely change. That's a product decision for @scottbuscemi 😄

Author: The [previous update](https://pantheon.io/docs/start-states/wordpress#2022-07-12) didn't necessarily align with a WordPress release (or, if it did, it wasn't noted), so I was following that precedent.

Author: In fact, none of the other updates in the release notes mention a WordPress version besides 6.0 (when themes broke), but I guess the change doesn't actually get applied until a WordPress release goes out.

Reviewer 2: just confirming that this PR is still good to go and that this convo here is "in the future"

🚀 👍 ?

Author: We actually chatted about this on our retro today. We need to have another conversation about it when @scottbuscemi is around, so let's hold off for now (unless there's a WordPress release that sneaks out that the change would go to users with).

#### <a name="20220816" class="release-update"></a>Deprecate `pantheon-cache` in favor of `pantheon cache`

> When the [Pantheon Advanced Page Cache](https://wordpress.org/plugins/pantheon-advanced-page-cache/) plugin is active, WordPress sites on Pantheon have two distinct but similarly-named WP-CLI commands: `wp pantheon cache` (which is provided by Pantheon Advanced Page Cache) and `wp pantheon-cache` (which is provided by the Pantheon mu-plugin. This creates in a confusing end result where it's unclear what Pantheon cache WP-CLI command to use or which subcommands are available for which parent commands.

Reviewer 2: ```suggestion
When the [Pantheon Advanced Page Cache](https://wordpress.org/plugins/pantheon-advanced-page-cache/) plugin is active, WordPress sites on Pantheon have two distinct but similarly-named WP-CLI commands: `wp pantheon cache` (which is provided by Pantheon Advanced Page Cache) and `wp pantheon-cache` (which is provided by the Pantheon mu-plugin. This leads to confusion about which Pantheon cache WP-CLI command to use or which subcommands are available for which parent commands.
```
